torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -316,21 +383,27 @@ void runKernel(
 // This function is here for testing purposes only
 void runTestKernel(
     CudaKernel* entry,
-    const at::ArrayRef<IValue>& inputs,
-    std::vector<at::Tensor>& outputs) {
+    const at::ArrayRef<IValue> inputs,
+    std::vector<at::Tensor> outputs) {
   const auto prior_device = at::cuda::current_device();
   at::cuda::set_device(entry->device_);
   auto stream = at::cuda::getCurrentCUDAStream();
 
   // TODO: Proper API to establish reasonable launch configurations;
   // Naive launch config;
+  TORCH_INTERNAL_ASSERT(!outputs.empty(), "No outputs set for test kernel.");
   size_t numel = outputs[0].numel();
 
   // TODO: we can't randomly clap down this until we got striding.
   const auto nBlocks = ceilDiv(numel, 128 * entry->unroll_factor_);
 
   KernelArgumentHolder kernel_args;
 
+  auto exprs = entry->outputs[0]->fusion()->exprs(true);
+  bool has_reduction = std::any_of(exprs.begin(), exprs.end(), [](Expr* expr) {
```

Comment:
NITPICK: there's a `has_random_` flag in fusion that marks whether or not a rand_like expression is there. Seems like a nice thing to have for reduction as well.